test-utils/src/commonMain/kotlin/mock/resources/MockResources.kt

```diff
@@ -0,0 +1,87 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.test.mock.resources
+
+import com.splendo.kaluga.base.text.format
+import com.splendo.kaluga.resources.Color
+import com.splendo.kaluga.resources.ColorLoader
+import com.splendo.kaluga.resources.Font
+import com.splendo.kaluga.resources.FontLoader
+import com.splendo.kaluga.resources.Image
+import com.splendo.kaluga.resources.ImageLoader
+import com.splendo.kaluga.resources.StringLoader
+
+class MockStringLoader private constructor (
+    private val string: String?,
+    private val quantityString: String?,
+    private val stringMap: Map<String, String?>,
+    private val quantityStringMap: Map<String, String?>
+): StringLoader {
+    constructor(string: String? = null, quantityString: String? = null):
+        this(string, quantityString, emptyMap(), emptyMap())
+    constructor(stringMap: Map<String, String?>, quantityStringMap: Map<String, String?>):
+        this(null, null, stringMap, quantityStringMap)
+
+    override fun loadString(identifier: String, defaultValue: String): String =
+        stringMap[identifier] ?: string ?: defaultValue
+
+    override fun loadQuantityString(
+        identifier: String,
+        quantity: Int,
+        defaultValue: String
+    ): String =
+        (quantityStringMap[identifier] ?: quantityString)?.format(quantity) ?: defaultValue
+}
+
+class MockColorLoader private constructor (
+    private val color: Color?,
+    private val colorMap: Map<String, Color?>
+): ColorLoader {
+    constructor(color: Color? = null): this(color, emptyMap())
+    constructor(colorMap: Map<String, Color?>): this(null, colorMap)
+
+    override fun loadColor(identifier: String, defaultValue: Color?): Color? =
+        colorMap[identifier] ?: color ?: defaultValue
+}
+
+class MockImageLoader private constructor (
+    private val image: Image?,
+    private val imageMap: Map<String, Image?>
+): ImageLoader {
+    constructor(image: Image? = null): this(image, emptyMap())
+    constructor(imageMap: Map<String, Image?>): this(null, imageMap)
+
+    override fun loadImage(identifier: String, defaultValue: Image?): Image? =
+        imageMap[identifier] ?: image ?: defaultValue
+}
+
+class MockFontLoader private constructor (
+    private val font: Font?,
+    private val fontMap: Map<String, Font?>
+): FontLoader {
+    constructor(font: Font? = null): this(font, emptyMap())
+    constructor(fontMap: Map<String, Font?>): this(null, fontMap)
+
+    override suspend fun loadFont(identifier: String, defaultValue: Font?): Font? =
+        fontMap[identifier] ?: font ?: defaultValue
+}
+
+expect fun mockColor(): Color
+expect fun mockImage(): Image
+// NOTE: returns null in Android unit tests
+expect fun mockFont(): Font?
```

Comment:
in Android unit tests both `Typeface.DEFAULT` and `Typeface.createFromFile(..)` return `null`. if someone knows a workaround to instantiate a Typeface instance please suggest

Reply:
For the Android unit tests Google assumes you will use a mocking library, I think Mockito is even already on the classpath.

I'd make sure to make different `actuals` for unit `test` and `androidTest` so on emulator you get a "real" font (this is done by using the correct sourcesets)

Reply:
fixed